lib/filter/filter-cmp.c

```diff
@@ -339,7 +339,7 @@ fop_cmp_clone(FilterExprNode *s)
 }
 
 FilterExprNode *
-fop_cmp_new(LogTemplate *left, LogTemplate *right, const gchar *type, gint compare_mode, EVTTAG *location)
+fop_cmp_new(LogTemplate *left, LogTemplate *right, const gchar *type, gint compare_mode, const gchar *location)
```

Comment:
I'm sure I am missing something, but I don't see how this is fixes a memory leak.

I can see that reusing the same `EVTTAG *` in two different `msg_warning()`s is a use after free, though, as `msg_event_free()` frees all the `EVTTAG *`s.

Could you kindly help me understand this change?

Reply:
Those messages are not always triggered. In fact neither of them are if you are running with `@version: 4.0`

In those cases the EVTTAG is leaked.

Reply:
Ooooh, I see, thanks!